oclmath/reference_math.cpp

```diff
@@ -71,15 +71,27 @@
 
 // Declare Classification macros for non-C99 platforms
 #ifndef isinf
-    #define isinf(x)    (	sizeof (x) == sizeof(float )	?	fabsf(x) == INFINITY  	\
-                        :	sizeof (x) == sizeof(double)	?	fabs(x) == INFINITY  	\
-                        :	fabsl(x) == INFINITY)
+#define isinf(x)                                    \
```

Comment:
Why not defining a function instead of using a macro calling a lambda?

Reply:
I didn't want to challange too many things. [`isinf`](https://en.cppreference.com/w/c/numeric/math/isinf) et al are supposed to be macros, and I didn't know if it would trigger allergic reactions if I changed that. The issue with the previous macro/implementation was that while the optimizer can always dead-code eliminate the branch of the select operator that's not taken, but the front-end still parses it and issues the diagnostic of implicit down-casting. `if constexpr` on the other hand is free of that.

Reply:
@keryell, are you okay to address this comment later?

Reply:
Sure. No problem.